storage/filesystem/object.go

```diff
@@ -187,6 +191,57 @@ func (s *ObjectStorage) encodedObjectSizeFromUnpacked(h plumbing.Hash) (
 	return size, err
 }
 
+func (s *ObjectStorage) packfile(idx idxfile.Index, pack plumbing.Hash) (*packfile.Packfile, error) {
+	if s.packfiles == nil {
+		if s.options.MaxOpenDescriptors > 0 {
```

Comment:
can you move this logic to a function?

Reply:
I've now split it up so the code that handles what is more clearly defined. At the start we check to see if the packfile is cached with `packfileFromCache`, at the end we do `storePackfileInCache` if a new packfile was opened.